### PR TITLE
KC-1603: Update versions to fix multiple CVEs

### PR DESCRIPTION
**Pull request by omkreddy**
Fixes some of the CVEs reported in https://confluentinc.atlassian.net/browse/KC-1603

io.confluent:confluent-log4j:1.2.17-cp2
httpclient: "4.5.13"
jetty: "9.4.40.v20210413"
jersey: "2.34"
mavenArtifact: "3.8.1"
reflections: "0.9.12"
zookeeper: "3.4.14"

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -78,18 +79,18 @@ versions += [
...
-  reflections: "0.9.11",
+  reflections: "0.9.12",
   rocksDB: "5.15.10",
   scalatest: "3.0.5",
   scoverage: "1.3.1",
   slf4j: "1.7.25",
   snappy: "1.1.7.2",
   zkclient: "0.11",
-  zookeeper: "3.4.13",
+  zookeeper: "3.4.14",
```
Comment (xjin-Confluent) [inline gradle/dependencies.gradle:93]: How about 3.5.9? I made this change in common.

**Comment by xjin-Confluent** [comment]
Please make similar changes to 2.3 and also ce-kafka. Thanks a lot.

**Review by xjin-Confluent — APPROVED** [review]

File: gradle/dependencies.gradle (hunk 1 above)
Comment (omkreddy) [reply gradle/dependencies.gradle:93]: I think it is risky to use 3.5.9 version in 5.2.x. I prefer to go with 3.4.14

Comment (xjin-Confluent) [reply gradle/dependencies.gradle:93]: BTW, https://confluentinc.atlassian.net/browse/KC-1703

**Pull request merged**